src/Arcus.WebApi.Security/Configuration/ArcusConfigurationProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using GuardNet;
+using Arcus.Security.Secrets.Core.Interfaces;
+
+namespace Arcus.WebApi.Security.Configuration
+{
+    /// <summary>
+    /// Provider to retrieve configuration tokens via an <see cref="ISecretProvider"/> implementation.
+    /// </summary>
+    internal class ArcusConfigurationProvider : ConfigurationProvider
+    {
+        private readonly ISecretProvider _secretProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ArcusConfigurationProvider"/> class.
+        /// </summary>
+        /// <param name="secretProvider">The provder to retrieve secret values for configuration tokens.</param>
+        internal ArcusConfigurationProvider(ISecretProvider secretProvider)
+        {
+            Guard.NotNull(secretProvider, nameof(secretProvider), $"Requires a {nameof(ISecretProvider)} instance");
+
+            _secretProvider = secretProvider;
+        }
+        
+        /// <summary>
+        /// Attempts to find a value with the given key, returns true if one is found, false otherwise.
+        /// </summary>
+        /// <param name="key">The key to lookup.</param>
+        /// <param name="value">The value found at key if one is found.</param>
+        /// <returns>True if key has a value, false otherwise.</returns>
+        public override bool TryGet(string key, out string value)
+        {
+            Task<string> getSecretValueAsync = _secretProvider.Get(key);
```

Comment:
Let's await it directly like we've discussed before:
```suggestion
            string secretValue = _secretProvider.Get(key).ConfigureAwait(false).GetAwaiter().GetResult();
```

No need to check for null here.

Reply:
No, what we discussed before was that we should use `async/await` even in overloaded methods. This extra check makes sure that we guard against miss-implemented `ISecretProvider` types.

```csharp
public Task<string> Get(string secretKey)
{
    return null;
{
```

Reply:
Then it would be ok to have an exception, but the current approach is also fine 👌